crates/sui-core/src/consensus_adapter.rs

```diff
@@ -212,17 +256,37 @@ impl ConsensusAdapter {
         // refreshed frequently enough to make sure this is Byzantine-resistant
     }
 
+    /// This method is called externally to begin reconfiguration
+    /// It transition reconfig state to reject new certificates from user
+    /// ConsensusAdapter will send EndOfPublish message once pending certificate queue is drained
+    pub fn begin_reconfiguration(self: &Arc<Self>) -> SuiResult {
+        let send_end_of_publish = {
+            let reconfig_guard = self.authority.get_reconfig_state_write_lock_guard();
+            self.authority.close_user_certs(reconfig_guard);
+            let pending_certificates = self.pending_certificates.lock();
+            pending_certificates.is_empty()
+        };
+        if send_end_of_publish {
+            if let Err(err) = self.submit(ConsensusTransaction::new_end_of_publish(
+                self.authority.name,
+            )) {
+                warn!("Error when sending end of publish message: {:?}", err);
```

Comment:
What happens next when this warning is hit?

Reply:
This is essentially storage error, so should not happen / or we can have panic here